signer/src/bin/main.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Offline transaction signer implementation
+//!
+//! WIP port / simplification from https://github.com/mobilecoinofficial/full-service/blob/fefe6f645d676b393ece2f607f0081304141b590/transaction-signer/src/bin/main.rs#L337
+
+use std::path::Path;
+
+use bip39::{Language, Mnemonic, MnemonicType};
+use clap::Parser;
+use log::{debug, info};
+use serde::{Deserialize, Serialize};
+
+use mc_core::{account::Account, slip10::Slip10KeyGenerator};
+use mc_crypto_ring_signature_signer::LocalRingSigner;
+use mc_transaction_core::AccountKey;
+use mc_transaction_signer::{read_input, write_output, Operations};
+
+#[derive(Clone, PartialEq, Debug, Parser)]
+struct Args {
+    /// Account secrets file
+    #[clap(long, short, default_value = "mc_secrets.json")]
+    secret_file: String,
+
+    #[command(subcommand)]
+    action: Actions,
+}
+
+#[derive(Clone, PartialEq, Debug, Parser)]
+enum Actions {
+    /// Create a new offline account, writing secrets to the output file
+    Create {
+        /// File name for account secrets to be written to
+        #[clap(short, long)]
+        output: String,
+    },
+    /// Import an existing offline account via mnemonic
+    Import {
+        /// Optional account name
+        #[clap(short, long)]
+        name: Option<String>,
+
+        /// Mnemonic for account import
+        mnemonic: String,
+
+        /// File for account secrets to be written to
+        #[clap(short, long)]
+        output: String,
```

Comment:
Similar question, when do you use this and when do you use `secret_file`?

Reply:
account_secrets_file is the input when running any of the signer commands, output is the output file to write the one created by the Create action

Reply:
Is there a reason why `Create` can't also use `account_secrets_file`?
It won't override an existing file so there's no risk of any damage happening.

Reply:
I find it more confusing but its not a big deal either way